### ISSUE TITLE
postgres migration cannot be run if schema is not defined in ormconfig

### ISSUE THREAD

**Issue by josephktcheung** [typeorm/typeorm#1820]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `websql`

**TypeORM version:**

[ ] `latest`
[x] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I'm using `typeorm@0.2.0-alpha.30`, here's my ormconfig;
```json
{
  "name": "default",
  "type": "postgres",
  "host": "localhost",
  "port": 1200,
  "username": "test",
  "password": "test",
  "database": "test",
  "entities": ["src/database/entity/**/*.ts"],
  "migrations": ["src/database/migration/**/*.ts"],
  "cli": {
    "entitiesDir": "src/database/entity",
    "migrationsDir": "src/database/migration"
  }
}
```

Here's my migration script:
```ts
import { MigrationInterface, QueryRunner } from 'typeorm';

export class AddUser1521967080143 implements MigrationInterface {
  public async up(queryRunner: QueryRunner): Promise<any> {
    await queryRunner.query(
      `CREATE TABLE "user" ("id" SERIAL NOT NULL, "firstName" character varying(100), "lastName" character varying(100), "facebookPSID" character varying, "email" character varying(100), "pictureId" character varying NOT NULL, CONSTRAINT "PK_cace4a159ff9f2512dd42373760" PRIMARY KEY ("id"))`
    );
  }

  public async down(queryRunner: QueryRunner): Promise<any> {
    await queryRunner.query(`DROP TABLE "user"`);
  }
}
```

When I run migration, following error occurs:
```
Error during migration run:
TypeError: Cannot read property 'indexOf' of undefined
    at PostgresQueryRunner.parseTableName (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/src/driver/postgres/PostgresQueryRunner.ts:1670:23)
    at PostgresQueryRunner.<anonymous> (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/src/driver/postgres/PostgresQueryRunner.ts:261:38)
    at step (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:42:23)
    at Object.next (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:23:53)
    at /Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:17:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:13:12)
    at PostgresQueryRunner.hasTable (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:343:16)
    at MigrationExecutor.<anonymous> (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/src/migration/MigrationExecutor.ts:210:46)
    at step (/Users/josephcheung/workspace/zeep/repos/zeep-user-service/graphql/node_modules/typeorm/migration/MigrationExecutor.js:32:23)
```

`typeorm@0.2.0-alpha.30` works if I add `schema` option to ormconfig and regenerate migration script with the schema like this:
```json
{
  "name": "default",
  "type": "postgres",
  "host": "localhost",
  "port": 1200,
  "username": "test",
  "password": "test",
  "database": "test",
  "schema": "public",
  "entities": ["src/database/entity/**/*.ts"],
  "migrations": ["src/database/migration/**/*.ts"],
  "cli": {
    "entitiesDir": "src/database/entity",
    "migrationsDir": "src/database/migration"
  }
}
```

```ts
import { MigrationInterface, QueryRunner } from 'typeorm';

export class AddUser1521967591012 implements MigrationInterface {
  public async up(queryRunner: QueryRunner): Promise<any> {
    await queryRunner.query(
      `CREATE TABLE "public"."user" ("id" SERIAL NOT NULL, "firstName" character varying(100), "lastName" character varying(100), "facebookPSID" character varying, "email" character varying(100), "pictureId" character varying NOT NULL, CONSTRAINT "PK_03b91d2b8321aa7ba32257dc321" PRIMARY KEY ("id"))`
    );
  }

  public async down(queryRunner: QueryRunner): Promise<any> {
    await queryRunner.query(`DROP TABLE "public"."user"`);
  }
}
```

I tried older version of `typeorm` (0.1.16) and it can run the migration successfully without schema defined in ormconfig.json.

**Comment by AlexMesser** [comment]
I tried to reproduce this issue on current `typeorm@0.2.0-alpha.39`, but migrations runs successfully. Can you try it on latest `@next` ? If you still got and error, please provide small repo with this problem.

**Comment by josephktcheung** [comment]
Hi @AlexMesser,

I've tested it again, I believe this is fixed since `typeorm@0.2.0-alpha.31`. This issue can be closed now.

Best,
Joseph
